ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    private Consumer<RequireSupport> toRequireSupport(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return requireSupport -> {
+            requireSupport.getLanguageInterpreter().ifPresent(builder::languageInterpreted);
+            requireSupport.getSignLanguageInterpreter().ifPresent(builder::signLanguageInterpreted);
+            requireSupport.getOtherSupport().ifPresent(builder::otherSupportRequired);
+
+            requireSupport.getHearingLoop()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::hearingLoop);
+
+            requireSupport.getDisabledAccess()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::disabledAccess);
+        };
+    }
+
+    private Consumer<ExpertRequest> toExpertRequest(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return expertRequest -> {
+            builder.expertEvidenceToExamine(expertRequest.getExpertEvidenceToExamine());
+            builder.reasonForExpertAdvice(expertRequest.getReasonForExpertAdvice());
+        };
+    }
+
+    private Consumer<Witness> toWitness(CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder) {
+        return witness -> {
+            builder.selfWitness(yesNoMapper.to(witness.getSelfWitness()));
+            witness.getNoOfOtherWitness().ifPresent(builder::numberOfOtherWitnesses);
+        };
+    }
+
+    private void toHearingLocation(
+        HearingLocation hearingLocation,
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        builder.hearingLocation(hearingLocation.getCourtName());
+        builder.hearingLocationSlug(hearingLocation.getHearingLocationSlug());
+        hearingLocation.getCourtAddress()
+            .ifPresent(address -> builder.hearingCourtAddress(addressMapper.to(address)));
+        builder.hearingLocationOption(CCDCourtLocationOption.valueOf(hearingLocation.getLocationOption().name()));
+        hearingLocation.getExceptionalCircumstancesReason().ifPresent(builder::exceptionalCircumstancesReason);
+    }
+
+    @Override
+    public DirectionsQuestionnaire from(CCDDirectionsQuestionnaire ccdDirectionsQuestionnaire) {
+        if (ccdDirectionsQuestionnaire == null) {
+            return null;
+        }
+
+        DirectionsQuestionnaire.DirectionsQuestionnaireBuilder builder = DirectionsQuestionnaire.builder();
+
+        builder.hearingLocation(extractHearingLocation(ccdDirectionsQuestionnaire));
+        builder.witness(extractWitness(ccdDirectionsQuestionnaire));
+        builder.expertRequest(extractExpertRequest(ccdDirectionsQuestionnaire));
+        builder.requireSupport(extractRequireSupport(ccdDirectionsQuestionnaire));
+
+        List<ExpertReport> expertReports = asStream(ccdDirectionsQuestionnaire.getExpertReports())
```

Comment:
Pls dont use asStream which is not implemented properly. The asStream in util needs to be removed or properly fixed. Its wrongly implemented

Reply:
Says who? 🤔 😸

Reply:
Says me.

Reply:
Looks fine to me. It's just NPE protection. If `getExpertReports()` returns `null` then `asStream` returns the empty stream and `expertReports` ends up as the empty list. Otherwise the correct stream is used and all is good...

What is your objection, @karthikkannankainos ?

Reply:
the asStream is generic but uses Collections.emptySet to return empty stream. Not correct. Also lacks unit tests for it. So request is to fix it and add unit test.
Fix is just change as single statement and keep it generic- ` return CollectionUtils.isEmpty(collection)? collection.stream():Stream.empty();`

Reply:
The suggested implementation is incorrect. as it is returning empty stream for non empty collection. 
```
return CollectionUtils.isEmpty(collection)? collection.stream():Stream.empty();
```

Reply:
Its a suggestion. So pls fix the if else condition properly. Thanks

Reply:
I agree that `asStream`should definitely be unit tested - but I don't see where it would be a bad implementation. It's returning an empty stream in case collection is null, which allows you to chain it with other functional methods without having to bubble the null check above.

Am I missing something?

Reply:
@simad the asStream is defined as generic of collection but the EmptySet is returned. So I suggested the change in my previous comments with the code change.